File: src/test/java/io/icednut/algorithm/exercise/graph/leet841/SolutionTest.java
package io.icednut.algorithm.exercise.graph.leet841;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class SolutionTest {

    @Test
    void test01() {
        var solution = new Solution();

        boolean output = solution.canVisitAllRooms(List.of(
                List.of(1),
                List.of(2),
                List.of(3),
                List.of()
        ));

        assertEquals(true, output);
    }

    @Test
    void test02() {
        var solution = new Solution();

        boolean output = solution.canVisitAllRooms(List.of(
                List.of(1, 3),
                List.of(3, 0, 1),
                List.of(2),
                List.of(0)
        ));

        assertEquals(false, output);
    }
}

File: src/test/kotlin/io/icednut/leetcode/exercise/singlenumber/SolutionTest.kt
package io.icednut.leetcode.exercise.singlenumber

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

internal class SolutionTest {

    @Test
    fun `싱글 넘버를 구할 수 있어야 한다`() {
        val solution = Solution()
        assertEquals(1, solution.singleNumber(intArrayOf(2, 2, 1)))
    }

    @Test
    fun `싱글 넘버를 구할 수 있어야 한다2`() {
        val solution = Solution()
        assertEquals(4, solution.singleNumber(intArrayOf(4,1,2,1,2)))
    }
}